auth.go

```diff
@@ -18,7 +19,15 @@ type AuthenticationResponse struct {
 	Scope        string `json:"scope"`
 }
 
-var cache = ttlcache.New[string, string]()
+var tokenCache cache.Cache
+
+func init() {
+	var err error
+	tokenCache, err = cache.NewCache("memory", `{}`)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Same as below, cache should not fail the execution, IMO.

Reply:
I think you are right. It's very unlikely to fail but better safe than sorry ! (eg: it would fail if we specified an adapter name for which there is no implementation, but we are providing an implementation for the cache "memory").